instrumentation/opentelemetry-instrumentation-tortoiseorm/src/opentelemetry/instrumentation/tortoiseorm/__init__.py

```diff
@@ -0,0 +1,316 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Instrument `tortoise-orm`_ to report SQL queries.
+Usage
+-----
+.. code:: python
+    from opentelemetry.instrumentation.tortoiseorm import TortoiseORMInstrumentor
+    from tortoise.contrib.fastapi import register_tortoise
+
+    TortoiseORMInstrumentor().instrument(tracer_provider=tracer)
+
+    register_tortoise(
+        app,
+        db_url=settings.db_url,
+        modules={"models": ["example_app.db_models"]},
+        generate_schemas=True,
+        add_exception_handlers=True,
+    )
+
+API
+---
+"""
+from typing import Collection
+
+import wrapt
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.tortoiseorm.package import _instruments
+from opentelemetry.instrumentation.tortoiseorm.version import __version__
+from opentelemetry.instrumentation.utils import unwrap
+from opentelemetry.semconv.trace import DbSystemValues, SpanAttributes
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCode
+
+try:
+    import tortoise.backends.asyncpg.client
```

Comment:
I'm not too familiar with `tortoise`, what determines whether or not these paths are importable or not?

Reply:
This is probably defined by a version number of the `tortoise` dependency. Please check this dependency instead of catching an error.

Reply:
Ah, I see. I tried a few things with `tortoise` and was able to do this:

``` python
In [1]: from tortoise import backends

In [2]: from pkgutil import iter_modules

In [3]: [module.name for module in iter_modules(backends.__path__)]
Out[3]: ['asyncpg', 'base', 'mysql', 'sqlite']
```